lib/Sema/SemanticPass/EnumValueResolver.hpp

```diff
@@ -0,0 +1,120 @@
+#ifndef GLU_SEMA_SEMANTICPASS_ENUMVALUERESOLVER_HPP
+#define GLU_SEMA_SEMANTICPASS_ENUMVALUERESOLVER_HPP
+
+#include "AST/ASTWalker.hpp"
+#include "Basic/Diagnostic.hpp"
+
+#include <llvm/ADT/APSInt.h>
+
+namespace glu::sema {
+
+/// @brief Resolves enum case values and ensures they have literal expressions.
+class EnumValueResolver : public ast::ASTWalker<EnumValueResolver, void> {
+    DiagnosticManager &_diagManager;
+
+    struct EnumReprInfo {
+        unsigned bitWidth = 32;
+        bool isUnsigned = false;
+    };
```

Comment:
We don’t even support negative integer literals, this seems unnecessary. All that matters is the bitwidth. Past Sema (which is where those values are used), only the bit pattern matters, not the signedness or the sign.